src/vision/mod.rs

```diff
@@ -79,7 +79,7 @@ macro_rules! feat_func_def {
                 let mut temp: i64 = 0;
                 let err_val = $ffi_name(&mut temp as MutAfArray, self.feat as Feat);
                 match err_val {
-                    0 => Ok(Array::from(temp)),
+                    0 => Ok(Array::from_retained(temp)),
```

Comment:
I don't think we need a new member function for this unless there are more use cases to it. Probably doing just `Ok(Array::from(temp).clone())` is enough. Can you please check it and change accordingly.